Sources/SwiftParser/IncrementalParseTransition.swift

```diff
@@ -345,29 +345,31 @@ public struct ConcurrentEdits: Sendable {
   }
 
   private static func translateSequentialEditsToConcurrentEdits(
-    _ edits: [IncrementalEdit]
-  ) -> [IncrementalEdit] {
-    var concurrentEdits: [IncrementalEdit] = []
+    _ edits: [SourceEdit]
+  ) -> [SourceEdit] {
+    var concurrentEdits: [SourceEdit] = []
     for editToAdd in edits {
       var editToAdd = editToAdd
       var editIndicesMergedWithNewEdit: [Int] = []
       for (index, existingEdit) in concurrentEdits.enumerated() {
-        if existingEdit.replacementRange.intersectsOrTouches(editToAdd.range) {
-          let intersectionLength =
-            existingEdit.replacementRange.intersected(editToAdd.range).length
+        if existingEdit.range.intersectsOrTouches(editToAdd.range),
+          let intersectionLength = existingEdit.range.intersecting(editToAdd.range)?.length
+        {
           let replacement: [UInt8]
           replacement =
-            existingEdit.replacement.prefix(max(0, editToAdd.offset - existingEdit.replacementRange.offset))
-            + editToAdd.replacement
-            + existingEdit.replacement.suffix(max(0, existingEdit.replacementRange.endOffset - editToAdd.endOffset))
-          editToAdd = IncrementalEdit(
+            existingEdit.replacementBytes.prefix(max(0, editToAdd.offset - existingEdit.replacementRange.offset))
+            + editToAdd.replacementBytes
+            + existingEdit.replacementBytes.suffix(
+              max(0, existingEdit.replacementRange.endOffset - editToAdd.endOffset)
+            )
+          editToAdd = SourceEdit(
             offset: Swift.min(existingEdit.offset, editToAdd.offset),
-            length: existingEdit.length + editToAdd.length - intersectionLength,
+            length: existingEdit.length + editToAdd.length - intersectionLength.utf8Length,
```

Comment:
Isn’t `intersectionLength` a `SourceLength` like the reset? So why do we need to go `.utf8Length` here?